autoload/black.vim

```diff
@@ -120,7 +135,12 @@ def _initialize_black_env(upgrade=False):
   return True
 
 if _initialize_black_env():
-  import black
+  # TODO: Better handling of the case when import succeeds but Black doesn't
```

Comment:
I don't understand this comment; if Black is uninstalled, surely the import would fail?

Reply:
Unfortunately, [not always](https://github.com/pypa/pip/issues/11835). This can be triggered by Vim built with Py3.12 importing black from `env/lib/python3.11/site-packages`, which this PR makes possible.
If black is uninstalled after it was used once in such a setup, then `pip uninstall` leaves `site-packages/black/__pycache__/` behind and that causes `import black` to succeed:
```
$ env/bin/python3 -c "import black; print(dir(black)); print(black.__file__)"
['__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__']
None
```
It's a corner case of a corner case, but it's a pain to troubleshoot when it bites.